framework/deproxy_client.py

```diff
@@ -187,7 +232,7 @@ def set_rps(self, rps):
         self.rps = rps
 
     def _stop_deproxy(self):
-        self.close()
+        self._do_close()
```

Comment:
You should move `_do_close` logic to `handle_close`  and remove `_do_close` method. It is unnecessary layer.

```python3
  def handle_close(self):
        self.close()
        self.writable = self._in_connecting_state
        self.handle_write = self.__setup_write
        self.conn_is_closed = True

   def _stop_deproxy(self):
       self.handle_close()
```